test/fuse/linux/read_test.cc

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/statfs.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <linux/fuse.h>
+
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "test/util/fuse_util.h"
+#include "test/util/test_util.h"
+
+#include "fuse_base.h"
+
+namespace gvisor {
+namespace testing {
+
+namespace {
+
+class ReadTest : public FuseTest {
+  void TearDown() { UnmountFuse(); }
+
+ protected:
+  const std::string test_file_ = "test_file";
+};
+
+TEST_F(ReadTest, ReadWhole) {
+  const std::string test_file_path =
```

Comment:
I suggest to declare this variable in protected area of the `ReadTest` testing suite since it is used across several tests.
```
  void SetUp() override {
    FuseTest::SetUp();
    test_file_path_ = JoinPath(mount_point_.path(), test_file_);
  }
```

Reply:
Might need reading multiple files in the future. Will consider cleaning up the test code after more tests are added (potentially after demo)

Reply:
FUSE_READ is not finished as a complete operation for demo purpose. Will add the missing ones (might need to change the fuse_base.cc) afterwards.